src/Elastic.Apm/Model/ExecutionSegmentCommon.cs

```diff
@@ -154,27 +161,45 @@ internal static Task<T> CaptureSpan<T>(Span span, Func<ISpan, Task<T>> func)
 					$"{nameof(Transaction)}.{nameof(CaptureException)}")
 			};
 
-			payloadSender.QueueError(new Error(ed, t.TraceId, t.Id, parentId ?? t.Id) { Culprit = capturedCulprit, Context = context });
+			payloadSender.QueueError(new Error(capturedException, executionSegment.TraceId, executionSegment.Id, parentId ?? executionSegment.Id,
+				transaction)
+			{
+				Culprit = capturedCulprit,
+				Context = context
+			});
 		}
 
-		public static void CaptureError(string message, string culprit, StackFrame[] frames, IPayloadSender payloadSender, IApmLogger logger,
-			IExecutionSegment t, Context context, string parentId = null
+		public static void CaptureError(
+			string message,
+			string culprit,
+			StackFrame[] frames,
+			IPayloadSender payloadSender,
+			IApmLogger logger,
+			IExecutionSegment executionSegment,
+			Context context,
```

Comment:
Should transaction's context always be used for error's context or context of the capturing execution segment?

Reply:
IIUC when capturing an error Java agent captures (by deep copy BTW) the context of the execution segment that captured the error - https://github.com/elastic/apm-agent-java/blob/ae57423b8dba1b8661344a3c2f43efa85e9fc658/apm-agent-core/src/main/java/co/elastic/apm/agent/impl/error/ErrorCapture.java#L124

Reply:
Yeah, as discussed, 
- we are not sure what `context` to use for `error`, since as it seems java also tries to use the `span's` context. As discussed let's just use the transaction's context for now and open follow up issues for 1) make sure we deep copy the transaction context 2) figure out if we maybe need to add a different `Context` type for this. Based on the [json schema](https://github.com/elastic/apm-server/blob/master/docs/spec/errors/error.json#L45) `Error.Context` seems to be the same as `Transaction.Context`, but it'd be worth asking other agent devs and verify.

Reply:
Done. Opened #191 and #192.